test/parallel/test-http-hostname-typechecking.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const vals = [{}, [], NaN, Infinity, -Infinity, true, false, 1, new Date()];
+
+vals.forEach((v) => {
+  assert.throws(
+    () => http.request({hostname: v}),
+    /^TypeError: "hostname" must either be a string, undefined or null$/);
+});
+
+['', undefined, null].forEach((v) => {
+  assert.doesNotThrow(() => {
+    http.request({hostname: v}).on('error', common.noop);
```

Comment:
Why the `common.noop` here? I'd think we'd either want `mustCall()` or `mustNotCall()`.